### ISSUE TITLE
[Subtask]: Support Spark 3.4.X

### ISSUE THREAD

**Issue by wangtaohz** [apache/amoro#1669]
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

Iceberg supports Spark 3.4 from version 1.3.0.

Arctic should also support Spark 3.4 version to facilitate Mixed Format users of Spark 3.4.

### Parent issue

#1554

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by zyuan2001** [comment]
I am interested in it, but my confusion is that there is a lot of duplicate code here, why not add a public module

**Comment by baiyangtx** [comment]
> I am interested in it, but my confusion is that there is a lot of duplicate code here, why not add a public module

@zyuan2001 yes, and there is an issue tracking this https://github.com/NetEase/arctic/issues/1449, are you interested to submit a PR to solve this?
